### PR TITLE
Update publishing.html.md

### PR DESCRIPTION
**Pull request by aicarmic**
Add OSS license requirement and list of permitted licenses to the instructions.

### CODE REVIEW

File: website/docs/registry/providers/publishing.html.md [hunk 1]
```diff
@@ -14,7 +14,23 @@ description: |-
 
 Providers published to the Terraform Registry are written and built in the same way as other Terraform Providers. For guidance on how to write a provider, see [Writing Custom Providers](/docs/extend/writing-custom-providers.html).
 
-The provider repository on GitHub must match the pattern `terraform-provider-{NAME}`, and the repository must be public.
+The provider repository on GitHub must match the pattern `terraform-provider-{NAME}`, and the repository must be public.  All Terraform providers must contain one of the following open source licenses:
```
Comment (justincampbell) [inline website/docs/registry/providers/publishing.html.md:17]: This looks good! If it was under another heading (h3 or h4) we could link directly to the license documentation from any errors we generate when a provider is publishing with an invalid license.

Comment (aicarmic) [reply website/docs/registry/providers/publishing.html.md:17]: good suggestion, I'll update with a distinct header

File: website/docs/registry/providers/publishing.html.md [hunk 2]
```diff
@@ -14,7 +14,25 @@ description: |-
 
 Providers published to the Terraform Registry are written and built in the same way as other Terraform Providers. For guidance on how to write a provider, see [Writing Custom Providers](/docs/extend/writing-custom-providers.html).
 
-The provider repository on GitHub must match the pattern `terraform-provider-{NAME}`, and the repository must be public.
+The provider repository on GitHub must match the pattern `terraform-provider-{NAME}`, and the repository must be public.
```
Comment (aicarmic) [inline website/docs/registry/providers/publishing.html.md:17]: @paultyng @justincampbell updated with separate header, and made it clear this requirement applies to partners only

File: website/docs/registry/providers/publishing.html.md [hunk 3]
```diff
@@ -14,7 +14,25 @@ description: |-
 
 Providers published to the Terraform Registry are written and built in the same way as other Terraform Providers. For guidance on how to write a provider, see [Writing Custom Providers](/docs/extend/writing-custom-providers.html).
 
-The provider repository on GitHub must match the pattern `terraform-provider-{NAME}`, and the repository must be public.
+The provider repository on GitHub must match the pattern `terraform-provider-{NAME}`, and the repository must be public.  
+
+#### Licensing a Provider
```
Comment (paultyng) [inline website/docs/registry/providers/publishing.html.md:19]:
```suggestion
#### Licensing a Verified Provider
```

**Comment by paultyng** [comment]
LGTM, one minor heading suggestion, but not required to merge.

**Pull request merged**